src/controllers/dlgprefcontroller.cpp

```diff
@@ -550,8 +551,21 @@ void DlgPrefController::slotApply() {
     }
 
     QString mappingPath = mappingPathFromIndex(m_ui.comboBoxMapping->currentIndex());
```

Comment:
Not your code, but it was confusing that the file name is also included. This is not the case if we use "path" in Qt like here: https://doc.qt.io/qt-6/qfileinfo.html#absolutePath 
Can you rename it to "mappingFilePath" along with the utility function?